cheroot/connections.py

```diff
@@ -267,14 +334,9 @@ def close(self):
             conn.close()
         self._readable_conns.clear()
 
-        connections = (
-            conn.data.socket
-            for _, conn in (self._selector.get_map() or {}).items()
-            if conn.data != self.server  # server closes its own socket
-        )
-        for connection in connections:
-            connection.close()
-
+        for (_, conn) in self._selector.connections:
+            if conn != self.server:  # server closes its own socket
```

Comment:
Using the identity check is probably better here:
```suggestion
            if conn is not self.server:  # server closes its own socket
```

Reply:
This suggestion was accepted inline.